code/modules/research/xenoarchaeology/tools/tools_anoscanner.dm

```diff
@@ -16,6 +16,17 @@
 	. = ..()
 	scan()
 
+/obj/item/device/ano_scanner/afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
```

Comment:
>`mob|obj|turf|area`

There's no point in this when you're changing the type afterwards `var/obj/effect/portal/P = target`